tests/BenchmarkDotNet.IntegrationTests/TailCallDiagnoserTests.cs

```diff
@@ -0,0 +1,103 @@
+
+        public TailCallDiagnoserTests(ITestOutputHelper outputHelper) : base(outputHelper)
+        {
+
+        }
+
+        public static IEnumerable<object[]> GetJits()
+            => new[]
+            {
+                new object[] { Jit.LegacyJit, Platform.X64, Runtime.Clr }, // 64bit LegacyJit for desktop .NET
+                new object[] { Jit.RyuJit, Platform.X64, Runtime.Clr }, // RyuJit for desktop .NET
+            };
+
+        public class TailCallBenchmarks
+        {
+            private long FactorialWithTailing(int pos, int depth)
+            {
+                return pos == 0 ? depth : FactorialWithTailing(pos - 1, depth * pos);
+            }
+
+            private long FactorialWithTailing(int depth)
+            {
+                return FactorialWithTailing(1, depth);
+            }
+            
+            [Benchmark]
+            public long Factorial()
+            {
+                return FactorialWithTailing(7);
+            }
+        }
+        
+        public class NonTailCallBenchmarks
+        {
+            private long FactorialWithoutTailing(int depth)
+            {
+                return depth == 0 ? 1 : depth * FactorialWithoutTailing(depth - 1);
+            }
+
+            [Benchmark]
+            public long Factorial()
+            {
+                return FactorialWithoutTailing(7);
+            }
+        }
+        
+        [Theory, MemberData(nameof(GetJits))]
+        [Trait(Constants.Category, Constants.BackwardCompatibilityCategory)]
+        public void TailCallDiagnoserCatchesTailCallEvents(Jit jit, Platform platform, Runtime runtime)
+        {
+            var output = Execute<TailCallBenchmarks>(jit, platform, runtime);
+            Assert.True(output.CapturedOutput.Where(x=>x.Text.Contains(TAIL_CALL_MARK)).Count() > 0);
+        }
+
+        [Theory, MemberData(nameof(GetJits))]
+        [Trait(Constants.Category, Constants.BackwardCompatibilityCategory)]
+        public void TailCallDiagnoserNotCatchesTailCallEvents(Jit jit, Platform platform, Runtime runtime)
+        {
+            var output = Execute<NonTailCallBenchmarks>(jit, platform, runtime);
+            Assert.True(output.CapturedOutput.Where(x => x.Text.Contains(TAIL_CALL_MARK)).Count() == 0);
+        }
+
+        private LogCapture Execute<T>(Jit jit, Platform platform, Runtime runtime)
+        {
+            var tailCallDiagnoser = new TailCallDiagnoser(false, true);
+            CanExecute<T>(CreateConfig(jit, platform, runtime, tailCallDiagnoser));
+            var output = (LogCapture)(tailCallDiagnoser as EtwDiagnoser<object>).GetType().GetField("Logger", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(tailCallDiagnoser);
+            return output;
+        }
+
+        private IConfig CreateConfig(Jit jit, Platform platform, Runtime runtime, TailCallDiagnoser diagnoser) => ManualConfig.CreateEmpty()
+            .With(Job.ShortRun.With(jit).With(platform).With(runtime))
```

Comment:
Am I right that we don't need actual benchmark results here? Could we use the dry job here (instead of `ShortRun`)?

Reply:
True, thanks for the advice